### ISSUE TITLE
Naming collision

### ISSUE THREAD

**Issue by fxhereng** [djchie/react-native-star-rating#1]
When I installed the lib, I had this error in my packager:

```
Error building DependencyGraph:
 Error: Naming collision detected: /home/cloud/React-native/ReactProject/node_modules/react-native-star-rating/node_modules/react-native/packager/react-packager/src/DependencyResolver/polyfills/polyfills.js collides with /home/cloud/React-native/ReactProject/node_modules/react-native/packager/react-packager/src/Resolver/polyfills/polyfills.js
    at HasteMap._updateHasteMap (HasteMap.js:123:13)
    at HasteMap.js:94:28
    at tryCallOne (/home/cloud/React-native/ReactProject/node_modules/react-native/node_modules/promise/lib/core.js:37:12)
    at /home/cloud/React-native/ReactProject/node_modules/react-native/node_modules/promise/lib/core.js:123:15
    at flush (/home/cloud/React-native/ReactProject/node_modules/react-native/node_modules/promise/node_modules/asap/raw.js:50:29)
    at doNTCallback0 (node.js:419:9)
    at process._tickCallback (node.js:348:13)
```

I tried to delete the react-native folder in the lib's node_modules but it didn't work.

I'm using rn 0.17.

Kind regards,

**Comment by thmohd** [comment]
Same Issue here

**Comment by kkgelu** [comment]
Same.

**Comment by d-a-n** [comment]
Is there any workaround for this?

**Comment by fxhereng** [comment]
@d-a-n The owner doesn't seem to be active, personally I found a workaround creating one myself. It's very basic but I can post the code if you're interested in.

**Comment by d-a-n** [comment]
@fxhereng That would be awesome. Thanks.

**Comment by fxhereng** [comment]
You need this library to be installed with MaterialIcons font: 

https://github.com/oblador/react-native-vector-icons

This is pretty basic, no animation but it can be a good start. I only tested it on Android.

`RatingStarsPicker.js`:

```javascript
'use strict';

const React = require('react-native');

const Icon = require('react-native-vector-icons/MaterialIcons');

const {
  View,
  TouchableOpacity,
} = React;

const styles = require('./styles/RatingStarsStyles');

/**
* Represents RatingStars picker
* @class RatingStarsPicker
* @constructor
* @module Main/views
*/
const RatingStarsPicker = React.createClass({

  propTypes: {
    initialRate: React.PropTypes.number,
    iconSize: React.PropTypes.number,
    color: React.PropTypes.string,
    style: View.propTypes.style,
  },

  getDefaultProps: function() {
    return {
      initialRate: 1,
      iconSize: 40,
      color: '#fff000',
    };
  },

  getInitialState: function() {
    return {
      ratesTab: this.getRatesTab(this.props.initialRate),
      rate: this.props.initialRate,
    };
  },

  /**
   * Get rates tab,
   * Example, a grade of 3.8 will be = [1, 1, 1, 0.8 ,0]
   * @param  {number} globalRate, rate to show out of 5 (ex: 3.8/5)
   * @return {array} result, array of each star grade to draw (between [0, 1])
   * @method getRatesTab
   */
  getRatesTab: function(globalRate) {
    let result = [0, 0, 0, 0, 0];

    if (globalRate === 0) {
      return result;
    }

    let floorRate = Math.floor(globalRate);
    let rest = globalRate - floorRate;

    for (let i = 0; i < floorRate; i++) {
      result[i] = 1;
    }

    result[floorRate] = rest;

    return result;
  },

  /**
   * On click star event
   * @param  {number} index of the star pressed
   * @method onClickStar
   */
  onClickStar: function(index) {
    this.setState({
      ratesTab: this.getRatesTab(index + 1),
      rate: index + 1,
    });
  },

  /**
   * Render a single star
   * @param  {number} starRate between [0,1]
   * @param  {number} index of the star 0, 1, 2, 3 or 4.
   * @return {View} star view
   */
  renderStar: function(starRate, index) {
    let starIconName = 'star-border'; //empty

    if (starRate === 1) {
      starIconName = 'star'; //full
    } else if (starRate > 0 && starRate < 1) {
      starIconName = 'star-half'; //Half
    }

    return (
      <TouchableOpacity style={this.props.style} activeOpacity={0.65} onPress={() => this.onClickStar(index)} >
        <Icon 
          name={starIconName} size={this.props.iconSize} 
          style={styles.starStyle} 
          color={this.props.color} />
      </TouchableOpacity>
    );
  },

  /**
  * Render the global view
  * 
  * @method render
  * @return {View} rating star picker view
  */
	render: function() {
		return (
      <View style={[styles.container, this.props.style]}>
        {this.renderStar(this.state.ratesTab[0], 0)}
        {this.renderStar(this.state.ratesTab[1], 1)}
        {this.renderStar(this.state.ratesTab[2], 2)}
        {this.renderStar(this.state.ratesTab[3], 3)}
        {this.renderStar(this.state.ratesTab[4], 4)}
      </View>
    );
  },
});

module.exports = RatingStarsPicker;

```

The styles I used: 

`RatingStarsStyles.js`:

```javascript
'use strict';

const React = require('react-native');
const {
  StyleSheet,
} = React;

/**
* Styles for the rating stars picker
*
* @property RatingStarsStyles
* @type StyleSheet
* @static
* @final
* @module Main/views/styles
*/
const RatingStarsStyles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  starStyle : {
    width: 40,
    height: 40,
  }
});

module.exports = RatingStarsStyles;

```

To use it:

```javascript
render: function() {
  <RatingStarsPicker
       initialRate={1}
       color={'your_color'}
       iconSize={star_size}
   />
}
```

**Comment by djchie** [comment]
Hey everyone, sorry for the delay. I just made a fix so update to 1.0.5 and it should work. Let me know if you guys run into any issues!
